src/app/vginventory/src/hooks/useUpdateVideoGame.ts

```diff
@@ -22,25 +25,56 @@ export function useUpdateVideoGame(vg: videoGame) {
         }
     };
 
-    function onDescriptionChange(event:any) {
-        if (event.target.value !== description) {
-            setDescription(event.target.value);
-        }
+    const addGenre = () => {
+        setGenres(genres => [...genres, ""]);
+    }
+
+    const updateGenre = (event:any, idx:number) => {
+        event.preventDefault();
+
+        const newGenres = genres.map((genre, sidx) => {
+            if (idx !== sidx) return genre;
+            return event.target.value;
+        });
+      
+        setGenres(newGenres);
+    }
+
+    const removeGenre = (idx:number) => {
+        setGenres(genres => genres.filter((g, sidx) => idx !== sidx));
+    }
+
+    const updateDescription = (event:any) => {
         event.preventDefault();
+        setDescription(event.target.value);
     }
 
-    function onUpdate(event:any) {
-        let game: videoGame = videogame;
-        game.description = description;
-        setVideoGame(game);
-        updateVideoGame();
+    const updateReleaseDate = (date:any) => {
+        setReleaseDate(date.toDate());
+    }
```

Comment:
this throws "date.toDate is not a function" exception you you empty the date field

Reply:
toDate is a moment obj function but if the date is not a valid moment object the onChange will return a string.